### PR TITLE
use parse_namelist

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR makes a bit more use of `parse_namelist`.

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -643,26 +586,9 @@ function EnvironmentVariables(namelist, Gr::Grid, param_set::PS) where {PS}
...
-        false
-    end
-
-    EnvThermo_scheme = try
-        string(namelist["thermodynamics"]["sgs"])
-    catch
-        println("Defaulting to saturation adjustment and microphysics with respect to environmental means")
-        "mean"
-    end
+    calc_tke = parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "calc_tke"; default = true)
```
Comment (yairchn) [inline src/types.jl:589]: I honestly think these flags (calc tke, calc_scalar_var) should be removed. we should always compute these variables. I should have removed them in the cleanup that I did

File: src/types.jl [hunk 2]
```diff
@@ -1013,110 +931,86 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
+        elseif entr_detr_fp_str == "dry"
+            entr_detr_dry
+        elseif entr_detr_fp_str == "inverse_w"
+            entr_detr_inverse_w
+        elseif entr_detr_fp_str == "b_w2"
             entr_detr_b_w2
-        end
-        if (calc_tke == false && "tke" in string(namelist["turbulence"]["EDMF_PrognosticTKE"]["entrainment"]))
-            error("Turbulence--EDMF_PrognosticTKE: >>calc_tke<< must be set to true when entrainment is using tke")
+        elseif entr_detr_fp_str == "entr_detr_tke"
```
Comment (yairchn) [inline src/types.jl:958]: this decision tree should be edited, many of these function are not used anymore and can be deleted

File: src/types.jl [hunk 3]
```diff
@@ -1013,110 +931,86 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
-                pressure_tan18_buoy
-            elseif string(namelist["turbulence"]["EDMF_PrognosticTKE"]["pressure_closure_buoy"]) == "normalmode"
-                pressure_normalmode_buoy
-            else
-                error("Turbulence--EDMF_PrognosticTKE: pressure closure in namelist option is not recognized")
-            end
-        catch
-            println("Turbulence--EDMF_PrognosticTKE: defaulting to pressure closure Tan2018")
+        pressure_func_buoy_str =
+            parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "pressure_closure_buoy"; default = "tan18")
```
Comment (yairchn) [inline src/types.jl:981]: we are not using this pressure function anymore, I think we can remove this if else command

**Review by yairchn — COMMENTED** [review]
not sure if my comments are related to the PR but some additional cleanup is needed

**Comment by charleskawczynski** [comment]
I've rebased after #72, so I think all of your (@yairchn) comments have been addressed.

File: src/types.jl [hunk 4]
```diff
@@ -935,82 +875,70 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
-                error("Turbulence--EDMF_PrognosticTKE: Entrainment rate namelist option is not recognized")
-            end
-        catch
-            println("Turbulence--EDMF_PrognosticTKE: defaulting to cloudy entrainment formulation")
-            entr_detr_b_w2
+        n_updrafts = parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "updraft_number"; default = 1)
+        use_const_plume_spacing =
+            parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "use_constant_plume_spacing"; default = false)
+
+        entr_detr_fp_str = parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "entrainment"; default = "b_w2")
```
Comment (yairchn) [inline src/types.jl:882]: the default should be "entr_detr_env_moisture_deficit"

File: src/types.jl [hunk 5]
```diff
@@ -935,82 +875,70 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
         end
 
         pressure_func_buoy = pressure_normalmode_buoy
+
+        pressure_func_drag_str = parse_namelist(
+            namelist,
+            "turbulence",
+            "EDMF_PrognosticTKE",
+            "pressure_closure_drag";
+            default = "tan18",
```
Comment (yairchn) [inline src/types.jl:902]: the default should be "normalmode"

File: src/types.jl [hunk 6]
```diff
@@ -935,82 +875,70 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
-        end
+        asp_label = parse_namelist(
+            namelist,
+            "turbulence",
+            "EDMF_PrognosticTKE",
+            "pressure_closure_asp_label";
+            default = "const",
+        )
+        similarity_diffusivity =
+            parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "use_similarity_diffusivity"; default = false)
```
Comment (yairchn) [inline src/types.jl:922]: this can be removed - the last cleanup removed similarity_diffusivity from the model

File: src/types.jl [hunk 7]
```diff
@@ -935,82 +875,70 @@ mutable struct EDMF_PrognosticTKE{PS, A1, A2}
...
+            parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "extrapolate_buoyancy"; default = true)
 
-        mixing_scheme = try
-            string(namelist["turbulence"]["EDMF_PrognosticTKE"]["mixing_length"])
-        catch
-            println("Using (Tan et al, 2018) default")
-            "default"
-        end
+        mixing_scheme =
+            parse_namelist(namelist, "turbulence", "EDMF_PrognosticTKE", "mixing_length"; default = "default")
```
Comment (yairchn) [inline src/types.jl:928]: this should be removed - there is only one mixing length scheme in the code now

**Review by yairchn — COMMENTED** [review]
LGTM - see the comments

**Review by haakon-e — APPROVED** [review]
nice solution

**Review by yairchn — APPROVED** [review]
LGTM

**Pull request merged**